purchase_request_operating_unit/wizard/purchase_request_line_make_purchase_order.py

```diff
@@ -22,7 +22,16 @@ def default_get(self, fields):
         res = super(PurchaseRequestLineMakePurchaseOrder, self).\
```

Comment:
I think the entire method could be simplified and made more efficient. Something like this:

```python
    @api.model
    def default_get(self, fields):
        res = super().default_get(fields)
        requests = self.env['purchase.request']
        active_model = self.env.context.get('active_model', False)
        active_ids = self.env.context.get('active_ids', [])
        active_recordset = self.env[active_model].browse(active_ids)
        if active_model == 'purchase.request.line':
            requests |= active_recordset.mapped('request_id')
        elif active_model == 'purchase.request':
            requests |= active_recordset
        operating_units = requests.mapped(operating_unit_id)
        if len(operating_units) >= 2:
            raise ValidationError(_('Could not process ! '
                                    'You have to select lines '
                                    'from the same operating unit.'))
        elif operating_units:
            res['operating_unit_id'] = operating_units.id
        return res
```

Reply:
I think it is a good opportunity to fix this method now. And do a forward port to 13.0.

Reply:
Sounds good for me too. Just let me know if I need to do something on my side, not an expert on GitHub per se.